src/VolumeRenderSettings.ts

```diff
@@ -0,0 +1,243 @@
+      const elem1 = a1[i];
+      const elem2 = a2[i];
+      if (elem1 instanceof Array && elem2 instanceof Array) {
+        if (!this.compareArray(elem1, elem2)) {
+          return false;
+        }
+      } else if (elem1 !== elem2) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Compares two VolumeRenderSettings objects.
+   * @returns true if both objects have identical settings.
+   */
+  public static isEqual(o1: VolumeRenderSettings, o2: VolumeRenderSettings): boolean {
+    for (const key of Object.keys(o1)) {
+      const v1 = o1[key];
+      const v2 = o2[key];
+      if (v1 instanceof Array) {
+        if (!this.compareArray(v1, v2)) {
+          return false;
+        }
+      } else if (v1 && v1.bmin !== undefined) {
+        // Bounds object
+        const bounds1 = v1 as Bounds;
+        const bounds2 = v2 as Bounds;
+        if (!bounds1.bmin.equals(bounds2.bmin) || !bounds1.bmax.equals(bounds2.bmax)) {
+          return false;
+        }
+      } else if (v1 instanceof Vector3 || v1 instanceof Vector2 || v1 instanceof Euler) {
+        if (!v1.equals(v2)) {
+          return false;
+        }
+      } else {
+        // number, boolean, string
+        if (v1 !== v2) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Recursively creates and returns a deep copy of an array.
+   * Note: assumes values in the array are either primitives (numbers) or arrays of primitives.
+   */
+  private static deepCopyArray(a: unknown[]): unknown[] {
+    const b: unknown[] = new Array(a.length);
+    for (let i = 0; i < a.length; i++) {
+      const val = a[i];
+      if (val instanceof Array) {
+        b[i] = this.deepCopyArray(val);
+      } else {
+        b[i] = val;
+      }
+    }
+    return b;
+  }
+
+  /**
+   * Creates a deep copy of a VolumeRenderSettings object.
+   * @param src The object to create a clone of.
+   * @returns a new VolumeRenderSettings object with identical fields that do not
+   * share references with the original settings object.
+   */
+  public static clone(src: VolumeRenderSettings): VolumeRenderSettings {
+    const dst = defaultVolumeRenderSettings(new Volume()); // initialize with empty volume
+    for (const key of Object.keys(src)) {
+      const val = src[key];
+      if (val instanceof Array) {
+        dst[key] = this.deepCopyArray(val);
+      } else if (key === "bounds") {
+        // must use key string here because Bounds is a type alias and not a class
+        dst.bounds.bmax = src.bounds.bmax.clone();
+        dst.bounds.bmin = src.bounds.bmin.clone();
+      } else if (val instanceof Vector3 || val instanceof Vector2 || val instanceof Euler) {
+        dst[key] = val.clone();
+      } else if (val instanceof String) {
+        dst[key] = "" + val;
+      } else {
+        // boolean, number, other primitives
+        dst[key] = val;
+      }
+    }
+    return dst as VolumeRenderSettings;
+  }
+}
```

Comment:
with all the code in this file, VolumeRenderSettings really looks more like a class so why not just make it one. Then all these functions can be member funcs in the class.  Is there any benefit to just keeping it as a type?

Reply:
Hmm, the only thing I can think of is making declaration of the `VolumeRenderSettings` objects a little more readable (object declaration with properties labeled), but considering almost all initialization is done through the current `getDefaultVolumeRenderSettings` method anyways it's the same as having the constructor set up default properties. Any thoughts?

Reply:
agree that constructor would be the default function, and clone would become a member function.   it's not a big problem -- we can leave things alone.  it's just one of those things where what you made is all the pieces of a class but it's not a class

Reply:
Okay, will do!